lib/src/main/java/ua/naiksoftware/stomp/StompClient.java

```diff
@@ -240,11 +240,12 @@ public Completable disconnectCompletable() {
             return Flowable.error(new IllegalArgumentException("Topic path cannot be null"));
         else if (!streamMap.containsKey(destPath))
             streamMap.put(destPath,
-                    subscribePath(destPath, headerList).andThen(
+                    Completable.defer(() -> subscribePath(destPath, headerList)).andThen(
```

Comment:
Defer the completable here so that if the subscription already exists another send request will not be made for the same destination.